### ISSUE TITLE
Preprocess hosts passed to "fab" cli before running tasks

### ISSUE THREAD

**Issue by brunson** [fabric/fabric#2244]
I think the last time I used fabric it was version 1.x, so the change to 3.0 is jump...

I have an inventory of about 19,000 hosts and it's written in an ansible inventory format yaml file. I would like to intercept the arguments to the `-H` option of fab and process that against the inventory to produce either a new iterable of hosts or an iterable of Connection objects via globbing, regexp, etc. That iterable would be used to drive the execution of tasks. 

I don't see an obvious way to do this in the documentation or a brief survey of the code for fabric or invoke. Any suggestions?

**Comment by davidjmemmett** [comment]
Fabric is v2, not v3 (fabric3 is an unmaintained, unofficial fork).

Fabric can be used as a library instead of a CLI tool, please [have a read at the documentation](https://docs.fabfile.org/en/stable/getting-started.html#multiple-servers) and see if it helps.

**Comment by brunson** [comment]
Yeah. I get it. Sorry for the error in naming. That fork is unfortunate and caused me many hours of frustration, I know it has nothing to do with you. The original fabric and the fab cli was yours, right?

Since posting this question I delved much deeper and saw the change in direction you went. I ended up writing my own cli based on your library which imitates a lot of the old fab utility from version 1. Thank you for making your project open source, it's been a boon to me twice now.

**Comment by kingbuzzman** [comment]
I'm in the same boat, the only viable method i see is overriding the `Executor.expand_calls`, but to do that I would need a new "executable" since i would be calling `fabric.main.make_program` directly and I feel like that is too much monkey patching. If anyone has a better solution PLEASE!! Document i there! Im going crazy!
